query.go

```diff
@@ -333,55 +333,67 @@ func (p *Page) Race() *RaceContext {
 	rc.branches = append(rc.branches, &raceBranch{
-		func() (*Element, error) { return rc.noSleepPage.Element(selector) },
-		callback,
+		condition: func() (*Element, error) { return rc.noSleepPage.Element(selector) },
+		callback:  func(e *Element) error { return nil },
 	})
 	return rc
 }
 
-// ElementX the doc is similar to ElementX but has a callback when a match is found
-func (rc *RaceContext) ElementX(selector string, callback func(*Element) error) *RaceContext {
+// ElementX the doc is similar to ElementX
+func (rc *RaceContext) ElementX(selector string) *RaceContext {
 	rc.branches = append(rc.branches, &raceBranch{
-		func() (*Element, error) { return rc.noSleepPage.ElementX(selector) },
-		callback,
+		condition: func() (*Element, error) { return rc.noSleepPage.ElementX(selector) },
+		callback:  func(e *Element) error { return nil },
 	})
 	return rc
 }
 
-// ElementR the doc is similar to ElementR but has a callback when a match is found
-func (rc *RaceContext) ElementR(selector, regex string, callback func(*Element) error) *RaceContext {
+// ElementR the doc is similar to ElementR
+func (rc *RaceContext) ElementR(selector, regex string) *RaceContext {
 	rc.branches = append(rc.branches, &raceBranch{
-		func() (*Element, error) { return rc.noSleepPage.ElementR(selector, regex) },
-		callback,
+		condition: func() (*Element, error) { return rc.noSleepPage.ElementR(selector, regex) },
+		callback:  func(e *Element) error { return nil },
 	})
 	return rc
 }
 
-// ElementByJS the doc is similar to MustElementByJS but has a callback when a match is found
-func (rc *RaceContext) ElementByJS(opts *EvalOptions, callback func(*Element) error) *RaceContext {
+// ElementByJS the doc is similar to MustElementByJS
+func (rc *RaceContext) ElementByJS(opts *EvalOptions) *RaceContext {
 	rc.branches = append(rc.branches, &raceBranch{
-		func() (*Element, error) { return rc.noSleepPage.ElementByJS(opts) },
-		callback,
+		condition: func() (*Element, error) { return rc.noSleepPage.ElementByJS(opts) },
+		callback:  func(e *Element) error { return nil },
 	})
 	return rc
 }
 
+// Handle adds a callback function to the most recent chained selector.
+// The callback function is run, if the corresponding selector is
+// present first, in the Race condition.
+func (rc *RaceContext) Handle(callback func(*Element) error) *RaceContext {
+	rc.branches[len(rc.branches)-1].callback = callback
+	return rc
+}
+
 // Do the race
-func (rc *RaceContext) Do() error {
-	return utils.Retry(rc.page.ctx, rc.page.sleeper(), func() (stop bool, err error) {
+func (rc *RaceContext) Do() (*Element, error) {
+	var el *Element
+	err := utils.Retry(rc.page.ctx, rc.page.sleeper(), func() (stop bool, err error) {
 		for _, branch := range rc.branches {
-			el, err := branch.condition()
+			bEl, err := branch.condition()
 			if err == nil {
-				return true, branch.callback(el)
+				el = bEl
+				return true, branch.callback(bEl)
```

Comment:
You can use `if branch.callback == nil` to detect if the callback is set or not, so you don't have to repeatly set empty ones `callback:  func(e *Element) error { return nil }`